csharp/device/Microsoft.Azure.Devices.Client/DeviceClient.cs

```diff
@@ -219,19 +250,34 @@ public AsyncTaskOfMessage ReceiveAsync(TimeSpan timeout)
         /// Deletes a received message from the device queue
         /// </summary>
         /// <returns>The lock identifier for the previously received message</returns>
+
+#if NETMF
```

Comment:
For all these methods that are entirely under if/else NETMF blocks I'm thinking about regrouping them all together into a single one, please let me know if you feel it would be a problem.